tests/common.py

```diff
@@ -49,8 +52,19 @@ def setUpClass(cls, test_class=None, years=None, years_non_observed=None):
         if getattr(test_class, "default_language") is not None:
             cls.set_language(test_class, test_class.default_language)
 
+        # Default `years_[insert]` to `years` to prevent redundant initialization.
+        years_islamic_no_estimated = years_islamic_no_estimated or years
+        years_non_observed = years_non_observed or years
+
         if years:
             cls.holidays = test_class(years=years)
+        if (
+            years_islamic_no_estimated
+            and "islamic_show_estimated" in signature(test_class.__init__).parameters
```

Comment:
I'd like to use `isinstance(cls.holidays, IslamicHolidays)`, but `cls.holidays` may not exist here. And if we make
`cls.holidays = test_class(years=years) if years else test_class()`
then there are problems in the tests of countries that don't have `years` in the initialization yet, and they need to be reworked...